osu.Game/Rulesets/Mods/ModDifficultyAdjust.cs

```diff
@@ -56,13 +57,14 @@ public override string SettingDescription
         {
             get
             {
-                string drainRate = DrainRate.IsDefault ? "" : $"HP {DrainRate.Value}";
-                string overallDifficulty = OverallDifficulty.IsDefault ? "" : $"OD {OverallDifficulty.Value}";
-
-                string[] settings = { drainRate, overallDifficulty };
-                // filter out empty strings so we don't have orphaned commas
-                settings = Array.FindAll(settings, s => !string.IsNullOrEmpty(s));
-                return string.Join(", ", settings);
+                string drainRate = DrainRate.IsDefault ? string.Empty : $"HP {DrainRate.Value:0.#}";
+                string overallDifficulty = OverallDifficulty.IsDefault ? string.Empty : $"OD {OverallDifficulty.Value:0.#}";
```

Comment:
you probably want to use `N1` instead of `0.#`, for better localisation support

https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings

Reply:
`N1` doesn't remove the trailing zero, is a tooltip like `Difficulty Adjust (CS 3.0) okay? The "details" tab for beatmaps doesn't show trailing zeroes either

Reply:
definitely

Reply:
matching the same formatting as the details display is fine.